scripts/qa_crowbarsetup.sh

```diff
@@ -563,6 +563,13 @@ function get_unclustered_sles12plus_nodes
     echo $sles12plusnodes
 }
 
+function get_non_controller_nodes
+{
+
+  sles12nodes=get_unclustered_sles12plus_nodes
+  controller_node=get_novacontroller
+  return ${sles12nodes[@]} ${controller_node[@]} | tr ' ' '\n' | sort | uniq -u
+}
```

Comment:
Not sure, but maybe there is a way to query the nodes that contains the nova-compute-* role via knife. Maybe in the logic the decides where to deploy Nova in other scenarios there is some clue.